Crash java.lang.ClassNotFoundException: ic2.core.item.tool.ContainerToolbox

### Your GTNH Discord Username

brandyyn

### Mod Version

bogosorter-1.2.25-GTNH

### Bug Report

trying to use bogosorter in my own pack and i crash becaused of "Caused by: java.lang.NoClassDefFoundError: ic2/core/item/tool/ContainerToolbox
	at com.cleanroommc.bogosorter.compat.DefaultCompat.init(DefaultCompat.java:209)"

[crash-2025-02-25_19.35.07-client.txt](https://github.com/user-attachments/files/18971417/crash-2025-02-25_19.35.07-client.txt)

### Java Version

Java 8

### Mod List or GTNH Pack Version

bogosorter-1.2.25-GTNH
modularui2-2.2.4-1.7.10

and i guess 
IC2Classic-1.7.10-1.2.5

### Final Checklist

- [x] I have searched the issues and haven't found a similar issue.

--- a/dependencies.gradle
+++ b/dependencies.gradle
@@ -81,6 +81,7 @@ dependencies {
     compileOnly rfg.deobf('curse.maven:terrafirmacraft-302973:2627990')
     compileOnly rfg.deobf('curse.maven:energy-control-373450:4776055')
     compileOnly rfg.deobf('curse.maven:industrial-craft-242638:2353971')
+    compileOnly rfg.deobf('curse.maven:ic2-classic-242942:5230413')
     compileOnly rfg.deobf('curse.maven:extra-utilities-225561:2264383')
     compileOnly rfg.deobf('curse.maven:hbms-nuclear-tech-mod-235439:6028711')
     compileOnly rfg.deobf('curse.maven:better-storage-232919:2248061')

--- a/src/main/java/com/cleanroommc/bogosorter/compat/DefaultCompat.java
+++ b/src/main/java/com/cleanroommc/bogosorter/compat/DefaultCompat.java
@@ -209,6 +209,10 @@ public class DefaultCompat {
             api.addCompat(ContainerToolbox.class, (container, builder) -> { builder.addSlotGroup(0, 9, 9); });
         }
 
+        if (IC2Classic.isLoaded()) {
+            api.addGenericCompat(ContainerPersonalChest.class);
+        }
+
         if (Bibliocraft.isLoaded()) {
             api.addCompat(
                 ContainerFramedChest.class,

--- a/src/main/java/com/cleanroommc/bogosorter/compat/loader/Mods.java
+++ b/src/main/java/com/cleanroommc/bogosorter/compat/loader/Mods.java
@@ -26,7 +26,16 @@ public enum Mods {
     GT6(() -> Loader.isModLoaded("gregtech") && Loader.isModLoaded("gregapi")),
     GalacticraftCore("galacticraftcore"),
     HBM("hbm"),
-    IC2("IC2"),
+    IC2(() -> Loader.isModLoaded("IC2") && !Loader.instance()
+        .getIndexedModList()
+        .get("IC2")
+        .getName()
+        .endsWith("Classic")),
+    IC2Classic(() -> Loader.isModLoaded("IC2") && Loader.instance()
+        .getIndexedModList()
+        .get("IC2")
+        .getName()
+        .endsWith("Classic")),
     ImmersiveEngineering("ImmersiveEngineering"),
     IronChest("IronChest"),
     Mekanism("Mekanism"),

